webdriver/tests/bidi/browsing_context/navigation_failed/navigation_failed.py

```diff
@@ -233,31 +234,51 @@ async def test_with_new_navigation(
     slow_page_url = url(
         "/webdriver/tests/bidi/browsing_context/support/empty.html?pipe=trickle(d10)"
     )
-    await subscribe_events(events=[NAVIGATION_FAILED_EVENT])
 
     result = await bidi_session.browsing_context.navigate(
         context=new_tab["context"], url=slow_page_url, wait="none"
     )
-    on_navigation_failed = wait_for_event(NAVIGATION_FAILED_EVENT)
+
+    # Depending on implementation, the `trickle(d10)` page can or can not yet
```

Comment:
isn't the slow URL ensures that no new document is created? or does it mean the slow URL navigation is flaky?

Reply:
As mentioned:
> When WPT trickles response, it sends headers and adds timeouts before sending the body. However, this does not guarantee the new document will not be created and activated.

[Response.write](https://github.com/web-platform-tests/wpt/blob/bf04f8ee54ee86605034d910a3e1d855ed7b6c36/tools/wptserve/wptserve/response.py#L239-L240) calls `write_status_headers` and then `write_content`. The `write_content` ends up in [`pipe.trickle.add_content`](https://github.com/web-platform-tests/wpt/blob/bf04f8ee54ee86605034d910a3e1d855ed7b6c36/tools/wptserve/wptserve/pipes.py#L256).

Reply:
how large is the timeout before the first chunks of the content are written? I'd prefer we change the test to ensure that the navigation is either aborted or cancelled. How about we use an endpoint that never writes a response body?

Reply:
Or perhaps not even the headers.

Reply:
> how large is the timeout before the first chunks of the content are written? I'd prefer we change the test to ensure that the navigation is either aborted or cancelled. How about we use an endpoint that never writes a response body?

The URL is `...?pipe=trickle(d10)`, which makes the body delay of 10 seconds. Meaning no body is sent before that timeout. The headers are sent fully. Changing the `pipe` mechanism would be great, but require an RFC. Until then IDK other way to test the difference between aborted and failed messags.

Reply:
In the scope of the test, 10 seconds equals to "never"

Reply:
Can we use the PythonScriptHandler handler?

Reply:
See webdriver/tests/support/http_handlers

Reply:
oh I guess it is only allowing synchronous handlers?

Reply:
so after checking the spec we confirmed it is possible for the document to be loaded while no body was received.

Reply:
So what if we use a normal page but with a slow loading resource like a javascript file? Then the load of the document should also be delayed. Or wouldn't this make a difference.

I tried to check the wpt.fyi results for this change for Firefox but I can only see the one for Safari which isn't helpful at all.

Reply:
I believe slow resources would not prevent the navigation from being committed. Basically as soon as the navigation request headers are received, the navigation in WebDriver BiDi can only be aborted and not cancelled, because as soon as the headers are read HTML spec allows a new document to be created (at least according to our reading of the spec).